lib/ui/form/form_screen.dart

```diff
@@ -1,44 +1,126 @@
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide FormState;
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:flutter_survey/di/di.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_survey/model/survey_details_model.dart';
+import 'package:flutter_survey/theme/app_dimensions.dart';
+import 'package:flutter_survey/ui/form/form_state.dart';
+import 'package:flutter_survey/ui/form/form_view_model.dart';
 import 'package:flutter_survey/ui/form/widget/form_survey_detail_page.dart';
 import 'package:flutter_survey/ui/form/widget/form_survey_question_page.dart';
+import 'package:flutter_survey/ui/widget/dimmed_background.dart';
+import 'package:flutter_survey/ui/widget/snack_bar.dart';
 import 'package:flutter_survey/usecases/get_survey_details_use_case.dart';
 
-class FormScreen extends StatelessWidget {
-  final String surveyId;
+final formViewModelProvider =
+    StateNotifierProvider.autoDispose<FormViewModel, FormState>((ref) {
+  return FormViewModel(getIt.get<GetSurveyDetailsUseCase>());
+});
 
-  final _pageController = PageController();
+final _surveyDetailsStreamProvider =
+    StreamProvider.autoDispose<SurveyDetailsModel>(
+        (ref) => ref.watch(formViewModelProvider.notifier).surveyDetails);
+
+final _errorStreamProvider = StreamProvider.autoDispose<String?>(
+    (ref) => ref.watch(formViewModelProvider.notifier).error);
 
-  // TODO: Integrate with VM #23
-  final _getSurveyDetailsUseCase = getIt.get<GetSurveyDetailsUseCase>();
+class FormScreen extends ConsumerStatefulWidget {
+  final String surveyId;
 
-  FormScreen({
+  const FormScreen({
     Key? key,
     required this.surveyId,
   }) : super(key: key);
 
+  @override
+  FormScreenState createState() => FormScreenState();
+}
+
+class FormScreenState extends ConsumerState<FormScreen> {
+  final _pageController = PageController();
+
+  @override
+  void initState() {
+    super.initState();
+    ref.read(formViewModelProvider.notifier).loadSurveyDetails(widget.surveyId);
+  }
+
   @override
   Widget build(BuildContext context) {
-    // TODO: Integrate with VM #23
-    _getSurveyDetailsUseCase.call(
-      GetSurveyDetailsInput(
-        surveyId: surveyId,
-      ),
-    );
+    final surveyDetails = ref.watch(_surveyDetailsStreamProvider).value;
+    final errorMessage = ref.watch(_errorStreamProvider).value ?? "";
+    return ref.watch(formViewModelProvider).when(
+          init: () => Container(),
```

Comment:
Is it necessary have an empty container? we can remove the `init state` completely, our first initial state can be the `loading state`. What do you think?